libp2p/stream/connection.nim

```diff
@@ -63,9 +70,29 @@ method initStream*(s: Connection) =
   inc getConnectionTracker().opened
 
 method close*(s: Connection) {.async.} =
-  await procCall LPStream(s).close()
-  inc getConnectionTracker().closed
+  if not s.isClosed:
+    await procCall LPStream(s).close()
+    inc getConnectionTracker().closed
 
 proc `$`*(conn: Connection): string =
   if not isNil(conn.peerInfo):
     result = conn.peerInfo.id
+
+func hash*(p: Connection): Hash =
+  cast[pointer](p).hash
+
+func `==`*(a, b: Connection): bool =
```

Comment:
no branch left behind 😄 
https://github.com/status-im/nim-libp2p/blob/6f638259ee81dc87078c17c12700ab5fb5d9feaa/libp2p/peerinfo.nim#L142-L156

Reply:
really unintuitive and error prone imho.

Reply:
It looks a little bit over-engineered. If you casting everything to pointers, you can simply do:
```nim
  cast[pointer](a) == cast[pointer](b)
```
In 3rd branch you are comparing pointers and then you are comparing values, but if pointers are equal values are equal too.